### ISSUE TITLE
Improved titles/custom result titles

### ISSUE THREAD

**Issue by ScriptPup** [mtxr/vscode-sqltools#892]
**Is your feature request related to a problem? Please describe.**
It's less of a problem and more of an annoyance. Many queries I run are verbose, so when I run them their titles are also extremely long which then pushes all other result sets off-screen and requires scrolling and hovering to find the one I'm looking for. I keep results sets up and 'alive' sometimes for days at a time if I'm working on something where I need data samples but don't need specific rows.

**Describe the solution you'd like**
I looked into custom titles and it seems like it's unlikely to be possible given how visual studio code handles them. That being said the investigation was very quick and dirty googling rather than checking the APIs. The most optimal solution would be to allow a user to right-click a result-set title and choose to rename it to whatever they like. (It'd also be awesome to be able to color code them, but I think that's even less likely to be possible!)

**Describe alternatives you've considered**
The more likely solution is more similar to what VSCode does natively now.  Instead of using the entire query as the title, we can just use the very first line and could even remove '--'. This way a user can write a comment above their code and the title will be based on the first line of text rather than the entire query. Example:
```
-- My query
SELECT 
     col1,
     col2,
     col3
FROM TABLE 
WHERE
    col4 = 'Something';
```
Would result in a title of `Query Con: My Query` instead of `Query Con: -- My Query SELECT col1, col2, col3,FROM TABLE WHERE col4 = 'Something';`

**Additional context**
My current workaround is to set titles to auto-collapse when there are too many on screen in .vs/settings.json
```
"workbench.editor.tabSizing": "shrink",
workbench.editor.pinnedTabSizing": "compact"
```

This isn't really optimal since I have to hover over each tab to find the results I'm looking for. In some cases this may be 10+ tabs so the time can stack up.

**Comment by Nintorac** [comment]
I would almost consider this a bug, I don't tend to keep queries around but usually have some windows open in side-by-side mode and after building a query I have to use my mouse's hyperscroll to go back to the beginning of the tabs list to close to the right (@ScriptPup's workaround will probably help with this however)

It looks to me like the title is being set here https://github.com/mtxr/vscode-sqltools/blob/ba294769be006e9e605d70ca487e5ce6fb4126c8/packages/plugins/connection-manager/webview/results.ts#L94-L111

If that's correct then I'm not sure why the query isn't being truncated to 16 chars already?

**Comment by gjsjohnmurray** [comment]
> If that's correct then I'm not sure why the query isn't being truncated to 16 chars already?

@Nintorac that change only shipped last month (August) in 0.24.0. Please update and confirm.

**Comment by gjsjohnmurray** [comment]
See also #635
